timesketch/lib/utils.py

```diff
@@ -68,6 +70,48 @@ def read_and_validate_csv(path):
             yield row
 
 
+def read_and_validate_jsonl(path):
+    """Generator for reading a JSONL (json lines) file.
+
+    Args:
+        path: Path to the JSONL file
+    """
+    # Fields that must be present in each entry of the JSONL file.
+    mandatory_fields = [u'message', u'datetime', u'timestamp_desc']
+
+    with open(path, 'rb') as fh:
+
+        lineno = 0
+        for line in fh:
+            lineno += 1
+            try:
+                linedict = json.loads(line)
+                if u'datetime' not in linedict.keys()\
```

Comment:
This logic is repeated in the read_and_vlidate_csv function. Can we refactor and break out the logic to it's own function and use that?

From read_and_vlidate_csv():

```
        for row in reader:
            if u'timestamp' not in csv_header and u'datetime' in csv_header:
                try:
                    parsed_datetime = parser.parse(row[u'datetime'])
                    row[u'timestamp'] = str(
                        int(time.mktime(parsed_datetime.timetuple())))
                except ValueError:
                    continue
```

Reply:
So, these similar, but actually different.  The current implementation of plaso json_lines output provides a timestamp, but not a datetime.  So we're working the other way here.  I've chatted with joachimmetz@ about adding it into that project, but for now we're turning a microsecond precision epoch timestamp into something ISO-formatted for use here.  Ultimately, because that's rather brittle, I opted to slice the first 10 intead of assume precision.  

Have I missed something?

Reply:
Ok, yeah it makes sense. Thanks for the explanation. You can proceed and discard this suggestion.